tests/test_async.py

```diff
@@ -24,6 +24,20 @@ async def test_wait_during_not_done(sleeper):
         sleeper.sleep_for_a_bit())
 
 
+@pytest.mark.asyncio
+async def test_wait_until_done_with_delay(sleeper):
+    await asyncio.gather(
+        wait_until_awake_with_delay(sleeper),
+        sleeper.sleep_for_a_bit())
+
+
+@pytest.mark.asyncio
+async def test_wait_until_done_with_too_much_delay(sleeper):
```

Comment:
I don't think you really need this test as you didn't touch any error handling code, can you make only this one fail?

Reply:
I feel better about having this test. It's another way of making sure that the asynchronous capabilities work as expected. It's short and simple, and generally speaking it's better to be on the safer side of things (it doesn't really cost us more).

Reply:
As I said, if you can make it fail it's good, otherwise it's just redundant code we need to maintain. the code was built this way so I would not have to test the same thing in different places